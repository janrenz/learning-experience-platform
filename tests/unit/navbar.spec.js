import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import HelloWorld from "@/components/NavBar.vue";
import BootstrapVue from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

// Use Vuex
localVue.use(Vuex);

// install plugins as normal
localVue.use(BootstrapVue);

describe("NavBar.vue", () => {
  let wrapper, store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        auth: {
          authenticated: false
        }
      }
    });
    wrapper = mount(HelloWorld, {
      localVue,
      store,
      stubs: ["router-link"]
    });
  });

  it("render when created ", () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should contain nav items ", () => {
    expect(wrapper.text()).toContain("About");
    expect(wrapper.text()).toContain("Courses");
    expect(wrapper.text()).toContain("Career");
  });

  it("should contain login button ", () => {
    const navLinks = wrapper.findAll("a");
    expect(navLinks.at(navLinks.length - 1).text()).toContain(
      "login / sign up"
    );
    expect(navLinks.at(navLinks.length - 1).attributes("role")).toContain(
      "button"
    );
  });
});
