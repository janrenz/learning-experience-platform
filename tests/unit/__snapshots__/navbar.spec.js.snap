// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue render when created  1`] = `
<div>
  <nav
    class="navbar lxp-navbar navbar-light navbar-expand-lg"
  >
    <div
      class="navbar-brand lxp-navbar-brand"
    >
      <router-link-stub
        to="/"
      >
        <img
          src="@/assets/whoa-logo-greyscale.svg"
        />
      </router-link-stub>
    </div>
     
    <button
      aria-controls="nav-collapse"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
     
    <transition-stub
      css="true"
      enteractiveclass="collapsing"
      enterclass=""
      entertoclass="collapse show"
      leaveactiveclass="collapsing"
      leaveclass="collapse show"
      leavetoclass="collapse"
    >
      <div
        class="navbar-collapse collapse"
        id="nav-collapse"
        style="display: none;"
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              <router-link-stub
                to="/about"
              >
                About
              </router-link-stub>
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              Courses
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              Career
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              Literature
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              Language Preference
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              <img
                alt="Search"
                height="18"
                src="@/assets/images/search.svg"
                width="18"
              />
            </a>
          </li>
           
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
              target="_self"
            >
              <a
                class="btn my-2 my-sm-0 ob-btn btn-outline-secondary btn-sm"
                href="#"
                id="lxp-signup"
                role="button"
                tabindex="0"
                target="_self"
              >
                login / sign up
          
              </a>
            </a>
          </li>
        </ul>
      </div>
    </transition-stub>
  </nav>
</div>
`;
